Sources/StrongContractResponse/All/StrongContract.handle.swift

```diff
@@ -43,29 +42,69 @@ public extension StrongContractClient.Request {
             // but the response body will not be sent to the client.
             app.get(pathComponents) {
                 if verbose { print("We received: \($0)") }
-                return try await payloadToResponse($0.codableBodyExemptingData(), $0).vaporResponse
+                return try await handler($0.decodedObject(), $0).vaporResponse
             }
         case .post:
             app.post(pathComponents) {
                 if verbose { print("We received: \($0)") }
-                return try await payloadToResponse($0.codableBodyExemptingData(), $0).vaporResponse
+                return try await handler($0.decodedObject(), $0).vaporResponse
             }
         case .put:
             app.put(pathComponents) {
                 if verbose { print("We received: \($0)") }
-                return try await payloadToResponse($0.codableBodyExemptingData(), $0).vaporResponse
+                return try await handler($0.decodedObject(), $0).vaporResponse
             }
         case .delete:
             app.delete(pathComponents) {
                 if verbose { print("We received: \($0)") }
-                return try await payloadToResponse($0.codableBodyExemptingData(), $0).vaporResponse
+                return try await handler($0.decodedObject(), $0).vaporResponse
             }
         case .patch:
             app.patch(pathComponents) {
                 if verbose { print("We received: \($0)") }
-                return try await payloadToResponse($0.codableBodyExemptingData(), $0).vaporResponse
+                return try await handler($0.decodedObject(), $0).vaporResponse
+            }
+        }
+    }
+
+    typealias Downloader = (Payload?, Vapor.Request) async throws -> Vapor.Response
```

Comment:
Not a fan of returning Vapor. Response: because that destroys the strong contract, the client could expect one type, but using this register could return an entirely different type in the body.

Reply:
How?

Reply:
What about the adaptor that retains type information is preventing the streaming of data?

Reply:
```
{ payload, request in
        let user = try await request.verifyAccessToken()
        guard let profilePicture = user.profilePicture else {
            throw Abort(.badRequest, reason: "no profile picture found")
        }
        let fileUrl = try request.fileUrl(for: user.strongId().uuidString + "/" + profilePicture)
        print("fileUrl path:", fileUrl.path)
        guard fileUrl.fileExists else {
            throw Abort(.notFound, reason: "File not found !!!")
        }
        return try await request.streamFile(at: fileUrl)
    }
```

Reply:
payload is also optional so you are not required to send the body

Reply:
You can do: "```swift" in the markdown.

Reply:
Oh I see.. the method you are calling, skips Data and goes directly to forming a response? (streamFile(at: fileUrl)?

Reply:
yeah we need to return a strem (byte chunks)

Reply:
Okay, then please put this all in a different extension: 

```swift 
extension 
```

Also please add a comment that `streamFile(at: )` requires us to break the strong contract.

Reply:
```swift 
public extension StrongContractClient.Request where Response == Data {
```